git-branchless-reword/src/lib.rs

```diff
@@ -62,9 +62,12 @@ pub enum InitialCommitMessages {
 #[instrument]
 pub fn edit_message(git_run_info: &GitRunInfo, repo: &Repo, message: &str) -> eyre::Result<String> {
     let mut editor = Editor::new();
-    let (editor, editor_program) = match get_editor(git_run_info, repo)? {
-        Some(editor_program) => (editor.executable(&editor_program), editor_program),
-        None => (&mut editor, "<default>".into()),
+    let editor_program = match get_editor(git_run_info, repo)? {
```

Comment:
As written after the change, this `match` expression both returns a value and has mutating side-effects, which can be surprising for further readers. I would prefer to keep this as is, or maybe move the `Editor` instantiation inside both cases of the `match` expression.

Reply:
Done - PTAL. I think b549e51 is around as verbose as the original code, but reads better (at least to me).

Reply:
I like it better too, thanks!